etna/models/sarimax.py

```diff
@@ -442,3 +459,59 @@ def __init__(
                 **self.kwargs,
             )
         )
+
+    @staticmethod
+    def _forecast_segment(
+        model, segment: Union[str, List[str]], ts: TSDataset, confidence_interval: bool = False
+    ) -> pd.DataFrame:
+        segment_features = ts[:, segment, :]
+        segment_features = segment_features.droplevel("segment", axis=1)
+        segment_features = segment_features.reset_index()
+        dates = segment_features["timestamp"]
+        dates.reset_index(drop=True, inplace=True)
+        segment_predict = model.predict(df=segment_features, confidence_interval=confidence_interval)
+        segment_predict = segment_predict.rename(
+            {"mean": "target", "mean_ci_lower": "target_lower", "mean_ci_upper": "target_upper"}, axis=1
+        )
+        segment_predict["segment"] = segment
+        segment_predict["timestamp"] = dates
+        return segment_predict
+
+    @log_decorator
+    def forecast(self, ts: TSDataset, confidence_interval: bool = False) -> TSDataset:
+        """Make predictions.
+        Parameters
+        ----------
+        ts:
+            Dataframe with features
+        confidence_interval:
+            If True returns confidence interval for forecast
+        Returns
+        -------
+        pd.DataFrame
+            Models result
+        Notes
+        -----
+        The width of the confidence interval is specified in the constructor of ProphetModel setting the interval_width
+        """
+        if self._segments is None:
+            raise ValueError("The model is not fitted yet, use fit() to train it")
+
+        result_list = list()
+        for segment in self._segments:
+            model = self._models[segment]
+
+            segment_predict = self._forecast_segment(model, segment, ts, confidence_interval)
```

Comment:
Have you considered possibility to use standard forecast method?

As I understand `confidence_interval` parameter makes it impossible if `_forecast_segment` is `staticmethod`, but it is not necessary to keep it static.